rmw_fastrtps_dynamic_cpp/CMakeLists.txt

```
@@ -100,49 +100,36 @@ add_library(rmw_fastrtps_dynamic_cpp
   src/type_support_registry.cpp
   src/rmw_get_endpoint_network_flow.cpp
 )
-target_link_libraries(rmw_fastrtps_dynamic_cpp
-  fastcdr fastrtps)
-
-# specific order: dependents before dependencies
-ament_target_dependencies(rmw_fastrtps_dynamic_cpp
-  "rcpputils"
-  "rcutils"
-  "rosidl_typesupport_fastrtps_c"
-  "rosidl_typesupport_fastrtps_cpp"
-  "rosidl_typesupport_introspection_c"
-  "rosidl_typesupport_introspection_cpp"
-  "rmw_dds_common"
-  "rmw_fastrtps_shared_cpp"
-  "rmw"
-  "rosidl_runtime_c"
+target_link_libraries(rmw_fastrtps_dynamic_cpp PUBLIC
+  fastcdr
+  fastrtps
+  rcpputils::rcpputils
+  rcutils::rcutils
+  rmw::rmw
+  rmw_fastrtps_shared_cpp::rmw_fastrtps_shared_cpp
+  rosidl_runtime_c::rosidl_runtime_c
+  rosidl_typesupport_fastrtps_c::rosidl_typesupport_fastrtps_c
+  rosidl_typesupport_fastrtps_cpp::rosidl_typesupport_fastrtps_cpp
+  rosidl_typesupport_introspection_c::rosidl_typesupport_introspection_c
+  rosidl_typesupport_introspection_cpp::rosidl_typesupport_introspection_cpp
 )
-
-target_link_libraries(rmw_fastrtps_dynamic_cpp
-  ${rmw_dds_common_LIBRARIES__rosidl_typesupport_introspection_cpp}
+target_link_libraries(rmw_fastrtps_dynamic_cpp PRIVATE
+  rmw_dds_common::rmw_dds_common_library
 )
 
 configure_rmw_library(rmw_fastrtps_dynamic_cpp)
 
 # Causes the visibility macros to use dllexport rather than dllimport,
 # which is appropriate when building the dll but not consuming it.
 target_compile_definitions(${PROJECT_NAME}
-PRIVATE "RMW_FASTRTPS_DYNAMIC_CPP_BUILDING_LIBRARY")
+  PRIVATE "RMW_FASTRTPS_DYNAMIC_CPP_BUILDING_LIBRARY")
 
 # Export old-style CMake variables
 ament_export_include_directories("include/${PROJECT_NAME}")
 ament_export_libraries(rmw_fastrtps_dynamic_cpp)
 
 # specific order: dependents before dependencies
-ament_export_dependencies(rcpputils)
-ament_export_dependencies(rcutils)
-ament_export_dependencies(rmw)
-ament_export_dependencies(rmw_dds_common)
-ament_export_dependencies(rmw_fastrtps_shared_cpp)
-ament_export_dependencies(rosidl_runtime_c)
-ament_export_dependencies(rosidl_typesupport_fastrtps_c)
-ament_export_dependencies(rosidl_typesupport_fastrtps_cpp)
-ament_export_dependencies(rosidl_typesupport_introspection_c)
-ament_export_dependencies(rosidl_typesupport_introspection_cpp)
+ament_export_dependencies(fastcdr rcpputils rcutils rmw rmw_fastrtps_shared_cpp rosidl_runtime_c rosidl_typesupport_fastrtps_c rosidl_typesupport_fastrtps_cpp rosidl_typesupport_introspection_c rosidl_typesupport_introspection_cpp)
```

Does this have to be a single line? that is kinda too long for a single line?

```suggestion
ament_export_dependencies(fastcdr
  rcpputils
  rcutils
  rmw
  rmw_fastrtps_shared_cpp
  rosidl_runtime_c
  rosidl_typesupport_fastrtps_c
  rosidl_typesupport_fastrtps_cpp
  rosidl_typesupport_introspection_c
  rosidl_typesupport_introspection_cpp
)
```